be/src/vec/exprs/table_function/vexplode_json_object.cpp

```diff
@@ -0,0 +1,159 @@
+        if (UNLIKELY(!doc || !doc->getValue())) {
+            // error jsonb, put null into output, cur_size = 0 , we will insert_default
+            return;
+        }
+        // value is NOT necessary to be deleted since JsonbValue will not allocate memory
+        JsonbValue* value = doc->getValue();
+        auto writer = std::make_unique<JsonbWriter>();
+        if (value->isObject()) {
+            _cur_size = value->length();
+            ObjectVal* obj = (ObjectVal*)value;
+            _object_pairs.first =
+                    ColumnNullable::create(ColumnString::create(), ColumnUInt8::create());
+            _object_pairs.second =
+                    ColumnNullable::create(ColumnString::create(), ColumnUInt8::create());
+            _object_pairs.first->reserve(_cur_size);
+            _object_pairs.second->reserve(_cur_size);
+            for (auto it = obj->begin(); it != obj->end(); ++it) {
+                _object_pairs.first->insert_data(it->getKeyStr(), it->klen());
+                writer->reset();
+                writer->writeValue(it->value());
+                if (it->value()->isNull()) {
+                    _object_pairs.second->insert_default();
+                } else {
+                    const std::string_view& jsonb_value = std::string_view(
+                            writer->getOutput()->getBuffer(), writer->getOutput()->getSize());
+                    _object_pairs.second->insert_data(jsonb_value.data(), jsonb_value.size());
+                }
+            }
+        }
+        // we do not support other json type except object
+    }
+}
+
+void VExplodeJsonObjectTableFunction::process_close() {
+    _json_object_column = nullptr;
+    _object_pairs.first = nullptr;
+    _object_pairs.second = nullptr;
+}
+
+void VExplodeJsonObjectTableFunction::get_same_many_values(MutableColumnPtr& column, int length) {
+    // if current is empty map row, also append a default value
+    if (current_empty()) {
+        column->insert_many_defaults(length);
+        return;
+    }
+    ColumnStruct* ret = nullptr;
+    // this _is_nullable is whole output column's nullable
+    if (_is_nullable) {
+        // make map kv value into struct
+        ret = assert_cast<ColumnStruct*>(
+                assert_cast<ColumnNullable*>(column.get())->get_nested_column_ptr().get());
+        assert_cast<ColumnUInt8*>(
+                assert_cast<ColumnNullable*>(column.get())->get_null_map_column_ptr().get())
+                ->insert_many_defaults(length);
+    } else if (column->is_column_struct()) {
+        ret = assert_cast<ColumnStruct*>(column.get());
+    } else {
+        throw Exception(ErrorCode::INTERNAL_ERROR,
+                        "only support map column explode to struct column");
+    }
+    if (!ret || ret->tuple_size() != 2) {
+        throw Exception(
+                ErrorCode::INTERNAL_ERROR,
+                "only support map column explode to two column, but given:  ", ret->tuple_size());
```

Comment:
it this message correct？ what's the relationship bettween this function and map type?

Reply:
it should be "only support expand json object int to kv pair column, but given:  "

Reply:
done!